prowler/lib/outputs/common.py

```diff
@@ -22,78 +23,93 @@ def get_provider_data_mapping(provider) -> dict:
     return data
 
 
-def generate_provider_output(provider, finding, csv_data) -> FindingOutput:
+def generate_output(provider, finding, output_options) -> Finding:
```

Comment:
What if we move this into the `Finding` class?

Please also add tests and a docsting to the function to explain inputs, outputs, and what it does.

Reply:
Done